### ISSUE TITLE
linter: jsx-a11y/anchor-has-content false positive with BaseUI render prop

### ISSUE THREAD

**Issue by heiwen** [oxc-project/oxc#18001]
### What version of Oxlint are you using?

1.39.0

### What command did you run?

oxlint

### What does your `.oxlintrc.json` config file look like?

```
{
  "$schema": "./node_modules/oxlint/configuration_schema.json",
  "plugins": ["eslint", "unicorn", "typescript", "oxc", "react", "promise", "jsx-a11y"],
  "categories": {
    "correctness": "error",
    "suspicious": "warn",
    "pedantic": "warn"
  },
  "rules": {
    "react-in-jsx-scope": "off",
    "max-lines-per-function": "off"
  }
}
```

### What happened?

BaseUI uses [render props ](https://base-ui.com/react/handbook/composition) to compose a Base UI part with your own React component.

For example, to render a `<a href... />` tag with a Button component, the syntax is:

```
<Button
  render={<a href="https://www.test.com" target="_blank" rel="noreferrer" />}
  nativeButton={false}
>
  CTA Text
</Button>
```

This incorrectly triggers the `eslint-plugin-jsx-a11y(anchor-has-content)` rule:

```
  × eslint-plugin-jsx-a11y(anchor-has-content): Missing accessible content when using `a` elements.
    ╭─[app/footer.tsx:20:27]
 19 │                 <Button
 20 │                   render={<a href="https://www.test.com" target="_blank" rel="noreferrer" />}
    ·                           ─────────────────────────────────────────────────────────────────────
 21 │                   nativeButton={false}
    ╰────
  help: Provide screen reader accessible content when using `a` elements.
```

Related tracking issue: https://github.com/oxc-project/oxc/issues/1141

**Comment by mehm8128** [comment]
ref: https://github.com/biomejs/biome/pull/10220
